### ISSUE TITLE
Settings reset to defaults upon updates when they shouldn't

### ISSUE THREAD

**Issue by Muehe** [Questie/Questie#3495]
[This comment](https://github.com/Questie/Questie/issues/3458#issuecomment-1121306569) seems to imply that settings are reverted to their defaults on updates, which shouldn't happen. Investigation is needed.

**Comment by BreakBB** [comment]
My two cents: Iirc from the past this was not only an issue on update, but a general update. And also not really an issue from Questie, but the game itself not saving/resetting SavedVariables.

**Comment by Muehe** [comment]
Yeah that happens. Ideally we should force a reload when we detect a setting migration/change, as it will persist the changes to disk.

But in this particular case the user had the setting changing back to the default and disabled it on every update, until the recent announce PR changed the default to off.

**Comment by Gogo1951** [comment]
> And also not really an issue from Questie, but the game itself not saving/resetting SavedVariables.

Not trying to be a dick, but I don't think this is accurate. Questie is the only add-on I have that has this problem. I have over 100 add-ons, and the only one that I've seen lose settings on update is Questie. Feels like this happens randomly every 3-4 patches.
